3.3/.sti/bin/assemble

```diff
@@ -1,23 +1,66 @@
 #!/bin/bash
 
-# For SCL enablement
-source .bashrc
+# This script uses these environment variables:
+#
+# DISABLE_COLLECTSTATIC: if not empty, inhibits execution of 'manage.py collectstatic'.
 
-set -e
+function assemble() {
+  # For SCL enablement
+  source .bashrc
 
-echo "---> Copying application source ..."
-cp -Rf /tmp/src/. ./
+  set -e
 
-if [ -f setup.py ]; then
-    echo "---> Installing application ..."
-    python setup.py install --user
-elif [ -f requirements.txt ]; then
-    echo "---> Installing dependencies ..."
-    pip install --user -r requirements.txt
-fi
+  echo "---> Copying application source ..."
+  cp -Rf /tmp/src/. ./
 
-# set permissions for any installed artifacts
-chmod -R og+rwx /opt/openshift
+  if [ -f requirements.txt ]; then
+      echo "---> Installing dependencies ..."
+      pip install --user -r requirements.txt
+  fi
 
-# remove pip temporary directory
-rm -rf /tmp/pip_build_default
+  if [ -f setup.py ]; then
+      echo "---> Installing application ..."
+      python setup.py install --user
+  fi
+
+  # set permissions for any installed artifacts
+  chmod -R og+rwx /opt/openshift
+
+  set +e
+  django_collectstatic
+  set -e
+
+  # remove pip temporary directory
+  rm -rf /tmp/pip_build_default
+}
+
+function django_collectstatic() {
+  [ -n "$DISABLE_COLLECTSTATIC" ] && return
+  ! pip show -q django && return
+
+  # Find shallowest manage.py script, either ./manage.py or <project>/manage.py
+  local manage_py=$(find . -maxdepth 2 -type f -name 'manage.py' -printf '%d\t%P\n' | sort -nk1 | cut -f2 | head -1)
+
+  if [ ! -f "$manage_py" ]; then
+    echo "WARNING: seems that you're using Django, but we could not find a 'manage.py' file."
+    echo "'manage.py collectstatic' ignored."
+    return
+  fi
+
+  echo "---> Collecting Django static files ..."
+  if ! python $manage_py collectstatic --dry-run --noinput &> /dev/null; then
```

Comment:
Look below... this is a dry run to check "collectability", only then we actually run it.
If you don't use static files, this command might always fail depending on your settings, and it doesn't mean the build should fail.

If you don't see your static files on the browser, you will see the message below in your build log, and from there it is better to run the command below on your local development until you've configured your project correctly.

What you do want in the build logs is the output of the second run (without the --dry-run flag).

Disclaimer: this is exactly what you get in Heroku.